ptxcompiler/patch.py

```diff
@@ -161,20 +167,37 @@ def patch_needed():
     else:
         check = True
 
-    if not check:
+    return check == False
```

Comment:
This is not quite right, since it turns `PTXCOMPILER_CHECK_NUMBA_CODGEN_PATCH_NEEDED=0` in the environment (which one would think would turn off the check) into `True`. You probably want `return not check`